src/util/expr.h

```diff
@@ -319,6 +319,40 @@ class exprt:public irept
   const_unique_depth_iteratort unique_depth_cend() const;
 };
 
+/// Protected class for all expressions.
+/// This API will eventually replace exprt.
+class protected_exprt : public exprt
+{
+protected:
+  // constructors
+  DEPRECATED("use protected_exprt(id, type) instead")
+  protected_exprt() : exprt()
+  {
+  }
+
+  DEPRECATED("use protected_exprt(id, type) instead")
+  explicit protected_exprt(const irep_idt &_id) : exprt(_id)
+  {
+  }
```

Comment:
It seems rather weird that newly introduced code would be deprecated? Can we just not have these?

Reply:
sure, removed

Reply:
Back again?